server/server.go

```diff
@@ -293,6 +304,19 @@ func (w *otherWorker) Do(job interface{}) interface{} {
 		w.s.stats.logRequestDuration(requestBegin)
 		return makeRespondResponse(pkt.ID, res)
 
+	case protocol.OpRPC:
+		codec := newServerCodec(pkt.Payload)
+
+		err := w.s.dispatcher.ServeRequest(codec)
+		if err != nil {
+			log.Errorf("Worker %v: ServeRPC: %v", w.name, err)
+			w.s.stats.logInvalid(requestBegin)
+			return makeErrResponse(pkt.ID, protocol.ErrInternal)
+		}
+
+		w.s.stats.logRequestDuration(requestBegin)
+		return makeRespondResponse(pkt.ID, codec.response)
+
```

Comment:
OH OK I think I understand now. Let me know if I've got this right: the client sends a message with the opcode OpRPC and whose payload is an encoded set of RPC arguments that will be interpreted by the `rpc` package. The server extracts the payload and passes it to the dispatcher, which in turn dispatches to a dynamically-registered RPC handler?